test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -2121,7 +2156,10 @@ public static void assertRows(UntypedResultSet result, Object[]... rows)
                 if (!((cellValidator == null && actualValue == null) || (cellValidator != null && cellValidator.equals(actualValue))))
                 {
                     Object actualValueDecoded = actualValue == null ? null : column.type.getSerializer().deserialize(actualValue);
-                    if (!Objects.equal(expected != null ? expected[j] : null, actualValueDecoded))
+                    Object expectedValueDecoded = expected != null ? expected[j] : null;
```

Comment:
honestly its been way too long I don't remember why this was done... this patch is from march... my brain struggles with last week =D